src/client/client.cpp

```diff
@@ -512,14 +513,12 @@ void Client::step(float dtime)
 			if (m_minimap && do_mapper_update)
 				m_minimap->addBlock(r.p, minimap_mapblock);
 
-			if (r.ack_block_to_server) {
-				/*
-					Acknowledge block
-					[0] u8 count
-					[1] v3s16 pos_0
-				*/
-				sendGotBlocks(r.p);
-			}
+			if (r.ack_block_to_server)
+				blocks_to_ack.emplace_back(r.p);
```

Comment:
Would it make sense to check if this vector holds more than 256 positions, and call `sendGotBlocks` right here?

i.e.

```cpp
if (r.ack_block_to_server) {
	if (blocks_to_ack.size() == 256) {
		sendGotBlocks(blocks_to_ack);
		blocks_to_ack.clear();
	}

	blocks_to_ack.emplace_back(r.p);
}
```

Reply:
Done, though with `255` not `256`.

Reply:
oops, yeah, 255 :)